sklearn/tree/tests/test_monotonic_tree.py

```diff
@@ -0,0 +1,235 @@
+                                 max_leaf_nodes=n_samples_train)
+        if hasattr(est, "random_state"):
+            est.set_params(**{"random_state": seed})
+        est.fit(X_train, y_train)
+        y = est.predict_proba(X_test)[:, 0]
+
+        # increasing constraint
+        y_incr = est.predict_proba(X_test_incr)[:, 0]
+        # y_incr should always be greater than y
+        assert np.all(y_incr >= y)
+
+        # decreasing constraint
+        y_decr = est.predict_proba(X_test_decr)[:, 0]
+        # y_decr should always be lower than y
+        assert np.all(y_decr <= y)
+
+
+def test_multiclass_raises():
+    X, y = datasets.make_hastie_10_2(n_samples=100, random_state=0)
+    y[0] = 0
+    monotonic_cst = np.zeros(X.shape[1])
+    monotonic_cst[0] = -1
+    monotonic_cst[1] = 1
+    for name, TreeClassifier in CLF_TREES.items():
+        est = TreeClassifier(max_depth=None, monotonic_cst=monotonic_cst,
+                             random_state=0)
+        if hasattr(est, "random_state"):
+            est.set_params(**{"random_state": 0})
+
+        with pytest.raises(ValueError):
+            est.fit(X, y)
+
+
+def assert_1d_reg_tree_children_monotonic_bounded(tree_, monotonic_sign):
+    # Flip values to always check for increasing constraint
+    values = monotonic_sign * tree_.value
+    for i in range(tree_.node_count):
+        if tree_.children_left[i] > i and tree_.children_right[i] > i:
+            # Check monotonicity
+            i_left = tree_.children_left[i]
+            i_right = tree_.children_right[i]
+            assert(float(values[i_left]) <= float(values[i_right]))
+            val_middle = float(values[i])
+            # Check bounds
+            if tree_.feature[i_left] >= 0:
+                i_left_right = tree_.children_right[i_left]
+                assert(float(values[i_left_right]) <= val_middle)
+            if tree_.feature[i_right] >= 0:
+                i_right_left = tree_.children_left[i_right]
+                assert(val_middle <= float(values[i_right_left]))
+
+
+def assert_1d_reg_monotonic(clf, monotonic_sign, min_x, max_x, n_steps):
+    X_grid = np.arange(min_x, max_x, (max_x-min_x)/n_steps).reshape(-1, 1)
+    y_pred_grid = clf.predict(X_grid)
+    assert (monotonic_sign * np.diff(y_pred_grid) >= 0.0).all()
+
+
+@pytest.mark.parametrize('monotonic_sign', (-1, 1))
+@pytest.mark.parametrize('splitter', ("best", "random"))
+@pytest.mark.parametrize('depth_first', (True, False))
+@pytest.mark.parametrize('seed', range(4))
+def test_1d_tree_nodes_values(monotonic_sign, splitter, depth_first, seed):
+    # Adaptation from test_nodes_values in test_montonic_constraints.py
+    # Build a single tree with only one feature, and make sure the nodes
+    # values respect the monotonic constraints.
+
+    # Considering the following tree with a monotonic POS constraint, we
+    # should have:
+    #
+    #       root
+    #      /    \
+    #     5      10
```

Comment:
I don't understand where 5 and 10 come from. Maybe name them c1 and c2? Or change the diagram to use a notation similar to `test_nd_tree_nodes_values`, as you wish.

Reply:
It's a just a remaining of the comment where I copied it (in sklearn/ensemble/_hist_gradient_boosting/tests/test_monotonic_contraints.py::test_nodes_values), indeed it's not relevant here.